### PR TITLE
upgrade django to support edx-analytics-data-api.

### PR DESCRIPTION
**Pull request by attiyaIshaque**
### [EDUCATOR-3274](https://openedx.atlassian.net/browse/EDUCATOR-3274)

To upgrade the version of Django in repo **[analytics-api](https://github.com/edx/edx-analytics-data-api/blob/master/requirements/base.in#L20)** we have to upgrade the version the **edx-enterprise-data**.

_Upgrade Django and all its dependencies._

@edx/educator-escalation Can you guys please review?

### CODE REVIEW

File: requirements/test-reporting.in [hunk 1]
```diff
@@ -1,6 +1,6 @@
 tox==3.0.0
 tox-battery==0.5.1
-pytest==3.5.1
+pytest==3.7.1
```
Comment (awaisdar001) [inline requirements/test-reporting.in:3]: Did you change that or the `make upgrade` did it?

Comment (attiyaIshaque) [reply requirements/test-reporting.in:3]: I have changed it. Changing the version of Django it also changed the version of pytest==3.7.1 see 
- https://github.com/edx/edx-enterprise-data/pull/51/files#diff-163afcbb9f302379161763ab1b938107R62
- https://github.com/edx/edx-enterprise-data/pull/51/files#diff-fed5f0d205a468ada128f1e04313fb62R19

Installing  the requirements it creates the conflicts of pytest. so I have to change into this file.

**Review by awaisdar001 — APPROVED** [review]

**Comment by awaisdar001** [comment]
Looks good to me!

**Review by noraiz-anwar — APPROVED** [review]
You would have to create a new release tag to use these changes in analytics-data-api.

**Comment by attiyaIshaque** [comment]
@noraiz-anwar okay, I will create it.

**Pull request merged**